src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -72,8 +72,29 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
 
             try
             {
+                ExportJobConfiguration exportJobConfiguration = _exportJobConfiguration;
+
+                string connectionHash = Microsoft.Health.Core.Extensions.StringExtensions.ComputeHash(_exportJobConfiguration.StorageAccountConnection);
+
+                if (string.IsNullOrEmpty(exportJobRecord.StorageAccountUri) && string.IsNullOrEmpty(_exportJobConfiguration.StorageAccountUri))
+                {
+                    if (!string.Equals(exportJobRecord.StorageAccountConnectionHash, connectionHash, StringComparison.Ordinal))
+                    {
+                        throw new DestinationConnectionException("Storage account connection string was updated during an export job.", HttpStatusCode.BadRequest);
+                    }
+                }
+                else
+                {
+                    if (!string.Equals(exportJobRecord.StorageAccountUri, _exportJobConfiguration.StorageAccountUri, StringComparison.Ordinal))
+                    {
+                        exportJobConfiguration = new ExportJobConfiguration();
+                        exportJobConfiguration.Enabled = _exportJobConfiguration.Enabled;
+                        exportJobConfiguration.StorageAccountUri = exportJobRecord.StorageAccountUri;
```

Comment:
Any reason we don't want to update `_exportJobConfiguration.StorageAccountUri` to the correct value and pass `_exportJobConfiguration` to `ConnectAsync()`?

Reply:
I was concerned that if we changed the default than it could be used in a different job, but this always checks the uri value in the job record so it wouldn't matter. I'll change it to update `_exportJobConfiguration`

Reply:
If the uri for the _exportJobConfiguration is changed here will it update the value elsewhere in the program, namely in the CreateExportRequestHandler? Looking at the documentation for IOption it is injected as a singleton, so both objects will share the same IOption for the export job configuration. I can't find anything saying if the value of an IOption is passed by reference or by value. I would assume by reference which would imply to me that updating it here would change it in CreateExportRequestHandler.